scripts/package/snap-get.sh

```diff
@@ -0,0 +1,286 @@
+        exit 1
+}
+
+# Get Snap version from remote repository
+is_available()
+{
+	# No error reporting when using -q
+	curl -s "${SNAP_REPO_HOST}${SNAP_VERSION_URL_PREFIX}$1.info"
+}
+
+# Check whether the snap package can be upgraded
+is_upgradeable()
+{
+	# Get snap info
+	FIND_SNAP=`snap list | grep $1 | awk '{$1=$1}1' OFS=" " || echo ""`
+
+	# Abort if not installed
+	if [ ! "${FIND_SNAP}" ] && [ ! "${FORCE_INSTALL}" ]; then
+		error "\nError: snap [$1] is not installed, can't upgrade. Use --force option."
+	fi
+
+	# Get current snap version
+	INSTVER="$(echo ${FIND_SNAP} | awk '{print $2}')"
+	# Get remote available version
+	REMVER="$(is_available $1)" || error "\nThe snap [$1] is not available in remote repo."
+	# Set new version in NEW_PACKAGE_VERSION var
+	NEW_PACKAGE_VERSION=${REMVER}
+
+	# Return success for upgrade or abort with up-to-date
+	# Note: sort -V, --version compare version strings
+	if [ "$(printf '%s\n' "$INSTVER" "$REMVER" | sort -V | head -n 1)" != "$REMVER" ]; then
+		return 0
+	else
+		logger -p local0.info -t "SnapUpdater[${$}]" "The snap [$1] is up to date (curr ver ${INSTVER}, remote ${NEW_PACKAGE_VERSION})"
+		error "\nThe snap [$1] is up to date (curr ver ${INSTVER}, remote ${NEW_PACKAGE_VERSION})"
+	fi
+}
+
+# Check whether the snap package is already installed 
+is_snap_installed()
+{
+	# Get snap info
+	FIND_SNAP=`snap list | grep $1 | awk '{$1=$1}1' OFS=", "`
+
+	# Abort if not installed and no force option
+	if [ "${FIND_SNAP}" ] && [ ! "${FORCE_INSTALL}" ]; then
+		error "Snap [$1] is already installed: ${FIND_SNAP}. Use --force option to reinstall."
+	fi
+
+	logger -p local0.info -t "SnapUpdater[${$}]" "Snap $1 is installed"
+}
+
+# Perform the Snap installation
+install_snap()
+{
+	PACKAGE_ARCH=$OS_ARCH
+	# Remote repository can use amd64 instead of x86_64
+	if [ "${PACKAGE_ARCH}" = "x86_64" ]; then
+		PACKAGE_ARCH="amd64"
+	fi
+	URL="${SNAP_REPO_HOST}${SNAP_PACKAGE_URL_PREFIX}$1_$(is_available $1)_$PACKAGE_ARCH.snap" ||
+		error "Error: the snap [$1], arch $OS_ARCH is not available for download."
+
+	if [ ! "${NEW_PACKAGE_VERSION}" ]; then
+		NEW_PACKAGE_VERSION=`echo ${URL} | awk -F'_' '{print $3}'`
+	fi
+
+	logger -p local0.info -t "SnapUpdater[${$}]" "New Snap [$1] version [${NEW_PACKAGE_VERSION}] is ready to be installed"
+
+	NAME="$(basename $URL)"
+	cd $WORKDIR
+
+	# Wget with -q doesn't report errors, we only handle the return code
+	curlt_code=0
+	curl -s --progress-bar -k --max-time 30 -o ${NAME} ${URL} || curl_code=1
+
+	if [ "${curl_code}" -ne 0 ]; then
+		echo "Error: failed to download snap [${NAME}] from repo"
+		logger -p local0.err -t "SnapUpdater[${$}]" "Error: failed to download snap [${NAME}] from repo"
+		return 1
+	fi
+
+	logger -p local0.info -t "SnapUpdater[${$}]" "New snap [$1] dowloaded, version ${NEW_PACKAGE_VERSION}"
+
+	# The snap install MUST be called by root: the user calling the utility must be in sudoers
+	sudo snap install ${DEVMODE} ./${NAME}
```

Comment:
We need to think very carefully about how sudo is going to work here. There is no requirement to have the foglamp user able to sudo at the moment.

We may need to consider an option such as setuid on execution script to get around this for example.